Flame graph category colors

gregtatum opened the pull request:
<s>Please only look at the last commit here.</s> (This is rebased on merged work now.)

@brisad this is a follow-up to the other PR. Feel free to look at it if you have time, otherwise wait until later :D

brisad commented:
I'll look at it later today :)

brisad commented on src/components/flame-graph/Canvas.js:
```diff
@@ -331,29 +190,24 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
+            ? colorStyles.selectedFillStyle
+            : colorStyles.unselectedFillStyle;
+        const foreground =
+          isHovered || isSelected ? colorStyles.selectedTextColor : '#000';
```
Below, `'HighlightText'` is still used for the foreground if an item is selected, causing bad contrast on the yellow boxes:
![screenshot_2019-01-16 perf html](https://user-images.githubusercontent.com/1911402/51278922-34bfdf00-19dc-11e9-9955-9d7913e544eb.png)

gregtatum replied:
Woops, that's definitely a mistake. I meant to use the shared styling. Thanks for finding it.

brisad commented on src/components/flame-graph/Canvas.js:
```diff
@@ -60,157 +60,6 @@ const ROW_HEIGHT = 16;
- * the lightness of the color. Darker colors indicate higher self
- * times.
- */
-export function getBackgroundColor(
```
By removing this, we no longer will see higher self-time encoded as a darker color. I thought it was neat as it made it easier to identify hot code. Don't we want that anymore?

gregtatum replied:
I'd like to move forward with the work here to consolidate the color categories, however I did file #1643 as a discussion issue around this change. I'm a bit sad with removing the work we did around coming up with this coloring scheme, but let's chat about this in that discussion issue.

brisad commented on src/components/flame-graph/Canvas.js:
```diff
@@ -383,74 +237,30 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
+        categories={categories}
+        durationText={`${(100 * duration).toFixed(2)}%`}
+        displayData={callTree.getDisplayData(callNodeIndex)}
+      />
```
This is a lot nicer! :+1:

brisad commented on src/components/stack-chart/Canvas.js:
```diff
@@ -382,72 +383,21 @@ class StackChartCanvas extends React.PureComponent<Props> {
+        callNodeIndex={callNodeIndex}
+        callNodeInfo={callNodeInfo}
+        categories={categories}
+        durationText={`${formatNumber(duration)}ms`}
```
I think we can use `formatMilliseconds` here instead.

gregtatum replied:
Whoops! Good call.

gregtatum commented on src/components/flame-graph/Canvas.js:
```diff
@@ -243,6 +92,15 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
     }
   }
 
+  // Re-use the colors from the ThreadActivityGraph,
```
Yeah, this comment feels outdated now. I updated it to:

```
  // Provide a memoized function that maps the category color names to specific color
  // choices that are used across this project's charts.
```

gregtatum commented:
@brisad this should be ready for re-review. I updated your comments, and fixed a small issue with the render function for the tooltip running too much because of the display data being recomputed every time.

gregtatum commented:
Woo! Thanks for the review.
